src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -170,12 +247,12 @@ private void ReportPendingWorkItemCount()
                     var reasons = workItem.InvocationReasons;
                     if (workItem.MustRefresh || reasons.Contains(PredefinedInvocationReasons.SyntaxChanged))
                     {
-                        await RunAnalyzersAsync(analyzers, document, (a, d, c) => a.AnalyzeSyntaxAsync(d, reasons, c), cancellationToken).ConfigureAwait(false);
+                        await RunAnalyzersAsync(analyzers, document, workItem, (a, d, c) => a.AnalyzeSyntaxAsync(d, reasons, c), cancellationToken).ConfigureAwait(false);
```

Comment:
nit: we can calculate applicable analyzers up front and reuse it for calls to `RunAnalyzersAsync`

Reply:
Yes, I thought about it, but then felt it is much cleaner and reliable to keep the computation of applicable analyzers at one place just prior to the core execution. If we see any perf issues, I will follow your suggestion.